flume-ng-core/src/test/java/org/apache/flume/source/TestAvroSource.java

```diff
@@ -129,6 +127,64 @@ public void testLifecycle() throws InterruptedException {
   }
 
   @Test
+  public void testSourceStoppedOnFlumeExceptionIfPortUsed()
+      throws InterruptedException, IOException {
+    final String loopbackIPv4 = "127.0.0.1";
+    final int port = 10500;
+
+    // create a dummy socket bound to a known port.
+    try (ServerSocketChannel dummyServerSocket = ServerSocketChannel.open()) {
+      dummyServerSocket.socket().setReuseAddress(true);
+      dummyServerSocket.socket().bind(new InetSocketAddress(loopbackIPv4, port));
+
+      Context context = new Context();
+      context.put("port", String.valueOf(port));
+      context.put("bind", loopbackIPv4);
+      // Invalid configuration may throw a FlumeException which has to be expected in the callers
+      Configurables.configure(source, context);
+      try {
+        source.start();
+        Assert.fail("Expected an exception during startup caused by binding on a used port");
+      } catch (FlumeException e) {
+        logger.info("Received an expected exception.", e);
+        Assert.assertTrue("Expected a bind related root cause",
```

Comment:
I kind of agree, but with the current implementation (throwing `new FlumeException(nce)`) the message itself is an implementation detail as well, because it comes directly from the netty code.
I'd suggest to explicitly define the message when throwing the `FlumeException`. Wdyt?